iModelCore/iModelPlatform/DgnCore/linestyle/LsDb.cpp

```diff
@@ -631,9 +638,69 @@ DgnStyleId LineStyleElement::ImportLineStyle(DgnStyleId srcStyleId, DgnImportCon
         return DgnStyleId();
         }
 
-    BentleyStatus result = importer.GetDestinationDb().LineStyles().Insert(dstStyleId, name.c_str(), compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetAttributes(jsonObj));
+    BentleyStatus result = importer.GetDestinationDb().LineStyles().Insert(dstStyleId, name.c_str(), compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetUnitDef(jsonObj));
     return (result == BSISUCCESS) ? dstStyleId : DgnStyleId();
     }
 
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+DgnElementPtr LineStyleElement::_CloneForImport(DgnDbStatus* status, DgnModelR destModel, DgnImportContext& context) const
+    {
+    DgnDbStatus _status;
+    if (nullptr == status)
+        status = &_status;
+
+
+    DgnElementPtr newElem = T_Super::_CloneForImport(status, destModel, context);
+    if (!newElem.IsValid() || (DgnDbStatus::Success != *status))
+        return newElem;
+
+    DgnStyleId srcStyleId = DgnStyleId(m_elementId.GetValue());
+
+    LineStyleElementCPtr srcStyle = LineStyleElement::Get(context.GetSourceDb(), DgnStyleId(srcStyleId));
+    BeAssert(srcStyle.IsValid());
+    if (!srcStyle.IsValid())
+        {
+        BeAssert(false && "invalid source line style ID");
+        return newElem;
+        }
+
+    DgnStyleId dstStyleId = QueryId(context.GetDestinationDb(), srcStyle->GetName().c_str());
+    if (dstStyleId.IsValid())
+        {
+        //  *** TBD: Check if the line style definitions match. If not, rename and remap
+        context.AddLineStyleId(srcStyleId, dstStyleId);
+        return const_cast<DgnElement*>(context.GetDestinationDb().Elements().GetElement(dstStyleId).get());
+        }
+
+    Utf8String name(srcStyle->GetName());
+    Utf8String  srcData (srcStyle->GetData());
+
+    Json::Value  jsonObj (Json::objectValue);
+    if (!Json::Reader::Parse(srcData, jsonObj))
+        return newElem;
+
+    LsComponentId compId = LsDefinition::GetComponentId (jsonObj);
+    compId = LsComponent::Import(compId, context);
+    if (!compId.IsValid())
+        {
+        //  *** TBD: Check if the line style definitions match. If not, rename and remap
+        BeAssert(false && "unable to import component for line style");
+        return newElem;
+        }
+
+    DefinitionModelPtr model = m_dgndb.Models().Get<DefinitionModel>(DgnModel::DictionaryId());
+
+    LsDefinition::InitializeJsonObject(jsonObj, compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetUnitDef(jsonObj));
+    Utf8String data = Json::FastWriter::ToString(jsonObj);
+
+    LineStyleElementPtr lsElement = dynamic_cast<LineStyleElement*>(newElem.get());
+    lsElement->SetName(name.c_str());
+    lsElement->SetDescription(nullptr);

```

Comment:
Name and description aren't changing. If someone added a description somehow, should leave it be...